### PR TITLE
chore(engine-core): Improve binary search logic

### PR DESCRIPTION
**Pull request by williamluke4**
The idea behind this is:

1. Make the code more readable
1. If the binary can not be found to display all the folders that it looked in. Not just the
contents of the last search space (Current)

### CODE REVIEW

**Comment by timsuchanek** [comment]
Let's merge this after the release, as this is critical logic

**Comment by timsuchanek** [comment]
@williamluke4 can you give some context what the improvements are for?

File: src/packages/engine-core/src/NodeEngine.ts [hunk 1]
```diff
@@ -320,10 +321,15 @@ You may have to run ${chalk.greenBright(
       this.currentRequestPromise.cancel()
     }
   }
-
-  private async resolvePrismaPath(): Promise<string> {
+  // TODO Clean Up Logic
```
Comment (timsuchanek) [inline src/packages/engine-core/src/NodeEngine.ts:324]: Do you still want to clean that up before merging?

File: src/packages/engine-core/src/NodeEngine.ts [hunk 2]
```diff
@@ -335,81 +341,52 @@ You may have to run ${chalk.greenBright(
 
     if (__filename.includes('NodeEngine')) {
       // TODO: Use engines package here
```
Comment (timsuchanek) [inline src/packages/engine-core/src/NodeEngine.ts:343]: TODO can be removed

File: src/packages/engine-core/src/NodeEngine.ts [hunk 3]
```diff
@@ -335,81 +341,52 @@ You may have to run ${chalk.greenBright(
...
+    }
+    const searchLocations: string[] = [
+      eval(`require('path').join(__dirname, '../../../.prisma/client')`), // Dot Prisma Path
+      this.generator?.output ?? eval('__dirname'), // Custom Generator Path
+      path.join(eval('__dirname'), '..'), // parentDirName
+      path.dirname(this.datamodelPath), // Datamodel Dir
+      this.cwd, //cwdPath
+    ]
+
+    for (let i = 0; i < searchLocations.length; i++) {
```
Comment (timsuchanek) [inline src/packages/engine-core/src/NodeEngine.ts:355]: Can you us a `for of` loop here instead?

File: src/packages/engine-core/src/NodeEngine.ts (hunk 1 above)
Comment (williamluke4) [reply src/packages/engine-core/src/NodeEngine.ts:324]: Will remove the comment

**Pull request merged**